.github/workflows/build-and-test.yml

```diff
@@ -52,6 +62,25 @@ jobs:
                   git fetch --prune --unshallow
                   git submodule -q update --init --recursive
 
+            - name: Setup DotNet SDK
```

Comment:
There's currently an [issue](https://github.com/actions/setup-dotnet/issues/136) with the setup-dotnet action which is preventing us from using it. 

The macOs and Ubuntu images already have the required SDK installed. 
https://github.com/actions/virtual-environments/blob/main/images/macos/macos-10.15-Readme.md#language-and-runtime

Reply:
I'm not sure that is a bug. In my projects I'm also had such problems with netcore 2.x and netcore 3.x.
I can propose one solution for this:
- Change build matrix
```yaml
matrix:
  framework:
    - 'netcoreapp2.1'
    - 'netcoreapp3.1'
    - 'net472'
```
- Configure matrix options:
```yaml
matrix:
  options:
    - os: ubuntu-latest
      runtime: -x64
      codecov: true
   - os: macos-latest
      runtime: -x64
      codecov: true
  - os: windows-latest
      runtime: -x64
      codecov: true

  exclude:
   - os: ubuntu-latest
      framework: 'net472'
   - os: macOS-latest
      framework: 'net472'
```
- Change runner:
```yaml
jobs:
  Build:
    name: ${{ matrix.options.os }} ${{ matrix.framework }}
    runs-on: ${{ matrix.options.os }}
```

After that - you'll have next builds:
```
macOS-latest netcoreapp2.1
macOS-latest netcoreapp3.1
ubuntu-latest netcoreapp2.1
ubuntu-latest netcoreapp3.1
windows-latest netcoreapp2.1
windows-latest netcoreapp3.1
windows-latest net472
```

And little change for `actions/setup-dotnet@v1`
```yaml
- name: Setup DotNet SDK 2.1.x
  if: (runner.os == 'macOS' || runner.os == 'Linux') && matrix.framework == 'netcoreapp2.1'
  uses: actions/setup-dotnet@v1
  with:
     dotnet-version: "2.1.x"

- name: Setup DotNet SDK 3.1.x
  if: (runner.os == 'macOS' || runner.os == 'Linux') && matrix.framework == 'netcoreapp3.1'
  uses: actions/setup-dotnet@v1
  with:
     dotnet-version: "3.1.x"
```

Reply:
I'd consider it a bug since it used to work well enough until recent updates. I also chatted to the repo owner on Twitter and he encouraged me to raise the issue. 

I don't understand the purpose of the `exclude:` additions here?

I'm not convinced your `dotnet-setup` filter will work. The application is multi targeted so if you setup `2.1.x` then that will update the path and the build will likely fail. (_Or does it work on unix for some bizarre reason?_)

What I'm trying to get across here is that we want the absolute minimum of changes in order to make this work. 

`macos-latest` on `3.1.x` is the only target we need for now and we don't want to be running the tests with code coverage on multiple platforms.

Reply:
exclude - means, CI will skip some build matrix options which you'd like to do not run.

`dotnet-setup` filter wil always work - it triggered by `if condition`. I agree that looks ugly, but it works explicitly.

I think, you can toying with all of these when you'll need. So, do you want to drop `setup-dotnet` action?
And... 6 tests on macOs failed, what I need to do with it? Because these fails not related to `libgdiplus`

Reply:
Thanks for the updates. The tests are failing because the test environment is expecting a different reference codec. We need to update the tests and reference codec registration to cater for the new environment.

https://github.com/SixLabors/ImageSharp/blob/120080b99f3f1be0fc4ecec6906ee683ea999a3d/tests/ImageSharp.Tests/TestUtilities/Tests/TestEnvironmentTests.cs#L84-L93

https://github.com/SixLabors/ImageSharp/blob/120080b99f3f1be0fc4ecec6906ee683ea999a3d/tests/ImageSharp.Tests/TestUtilities/TestEnvironment.Formats.cs#L53-L77